### PR TITLE
Add documentation for to_char and the specification language

### PR DESCRIPTION
**Pull request by quodlibetor**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4882)

### CODE REVIEW

File: doc/user/content/sql/functions/to_char.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: "to_char Function"
+description: "Converts a timestamp into a string using the specified format."
+menu:
+  main:
+    parent: 'sql-functions'
+---
+
+`to_char` returns a string created using the specified format string.
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:9]: Let's make this lead-in look more like the description, e.g.

> `to_char` converts a timestamp into a string using the specified format.

File: doc/user/content/sql/functions/to_char.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+The format string can be composed of any number of format tokens, interspersed with
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:11]: These are referred to as "format tokens" here but "format specifiers" below; let's make those consistent and introduce links between this reference and the header where they can be found.

File: doc/user/content/sql/functions/to_char.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+| `HH`                        | hour of day (01-12)                                                                              |
+| `HH12`                      | hour of day (01-12)                                                                              |
+| `HH24`                      | hour of day (00-23)                                                                              |
+| `MI`                        | minute (00-59)                                                                                   |
+| `SS`                        | second (00-59)                                                                                   |
+| `MS`                        | millisecond (000-999)                                                                            |
+| `US`                        | microsecond (000000-999999)                                                                      |
+| `SSSS`                      | seconds past midnight (0-86399)                                                                  |
+| `AM`/`am`/`PM`/`pm`         | AM, am, PM, or pm meridiem indicator (without periods)                                           |
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:63]: The description for this leaves me with some questions about what this does, e.g. are the meridiem indicators here case-sensitive? If not maybe stress that the pattern is case insensitive and does not accept periods.

File: doc/user/content/sql/functions/to_char.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+| `A.M.`/`a.m.`/`P.M.`/`p.m.` | A.M., a.m., P.M., or p.m. meridiem indicator (with periods)                                      |
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:64]: Same thing here

File: doc/user/content/sql/functions/to_char.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  formatted
+ ------------
+  6th of Nov
+```
+
+## Format Specifiers
+
+Regular specifiers:
+
+| Pattern                     | Description                                                                                      |
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:53]: "Pattern" here another name for these things. Seems like having one consistent name would make this nominally easier to reference.

File: doc/user/content/sql/functions/to_char.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+| `IW`                        | week number of ISO 8601 week-numbering year (01-53; the first Thursday of the year is in week 1) |
+| `CC`                        | century (2 digits) (the twenty-first century starts on 2001-01-01)                               |
+| `J`                         | Julian Day (days since November 24, 4714 BC at midnight)                                         |
+| `Q`                         | quarter (ignored by to_date and to_timestamp)                                                    |
+| `RM`                        | month in upper case Roman numerals (I-XII; I=January)                                            |
+| `rm`                        | month in lower case Roman numerals (i-xii; i=January)                                            |
+| `TZ`                        | upper case time-zone name                                                                        |
+| `tz`                        | lower case time-zone name                                                                        |
+
+Specifier Modifiers:
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:105]: Maybe make this an `<h3>`? and ensure that the word "specifier" here is in sync with the name elsewhere.

File: doc/user/content/sql/functions/to_char.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+```
+```nofmt
+  formatted
+ ------------
+  6th of Nov
+```
+
+## Format Specifiers
+
+Regular specifiers:
```
Comment (sploiselle) [inline doc/user/content/sql/functions/to_char.md:51]: I think we can elide this `<p>`.

**Review by sploiselle — APPROVED** [review]
LGTM; have some wording/heading suggestions but I'm good to merge this with a first attempt at solving those and can iterate if folks are confused.

File: doc/user/data/sql_funcs.yml [hunk 8]
```diff
@@ -278,6 +278,12 @@
     description: Converts Unix epoch (seconds since 00:00:00 UTC on January 1, 1970)
       to timestamp
 
+  - signature: 'to_char(val: timestamp, format: str)'
+    description: >-
+      Converts a timestamp into a string using the specified format, see the
+      detailed docs for the full specification language.
```
Comment (benesch) [inline doc/user/data/sql_funcs.yml:284]: I think we can drop the "see the detailed docs for the full specification language" clause for consistency with the other descriptions that have a more detailed URL. (Also it's a run-on sentence.)

File: doc/user/content/sql/functions/to_char.md (hunk 6 above)
Comment (benesch) [reply doc/user/content/sql/functions/to_char.md:105]: Also headers should use sentence case, so "Specifier modifiers"

File: doc/user/content/sql/functions/to_char.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+---
+
+`to_char` returns a string created using the specified format string.
+
+The format string can be composed of any number of format tokens, interspersed with
+regular text. You can place a format token inside of double-quotes to emit it literally.
+
+## Examples
+
+**RFC 2822 Format**
```
Comment (benesch) [inline doc/user/content/sql/functions/to_char.md:16]: This seems like a heading; it should probably just be a heading? Also should be sentence case.

File: doc/user/content/sql/functions/to_char.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+```sql
+SELECT to_char(TIMESTAMPTZ '2019-11-26 15:56:46 +00:00', 'Dy, Mon DD YYYY HH24:MI:SS +0000') AS formatted
+```
+```nofmt
+             formatted
+ ---------------------------------
+  Tue, Nov 26 2019 15:56:46 +0000
+```
+
+**Additional non-interpreted Text**
```
Comment (benesch) [inline doc/user/content/sql/functions/to_char.md:27]: Ditto on sentence case.

File: doc/user/content/sql/functions/to_char.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+```sql
+SELECT to_char(TIMESTAMPTZ '2019-11-26 15:56:46 +00:00', '"Welcome" to Mon, YYYY') AS formatted
+```
+```nofmt
+       formatted
+ ----------------------
+  Welcome to Nov, 2019
+```
+
+**Ordinal Modifiers**
```
Comment (benesch) [inline doc/user/content/sql/functions/to_char.md:38]: Idem.

File: doc/user/content/sql/functions/to_char.md (hunk 3 above)
Comment (benesch) [reply doc/user/content/sql/functions/to_char.md:63]: The Postgres docs just say "meridiem indicator (without periods)", which I actually think is more clear despite being more terse. The phrasing here kind of makes it seem like if you write `pm` you'll only ever get an "pm" as output. Maybe we should split it out into all four cases?

```
`AM` or `PM` | uppercase meridiem indicator without periods
`am` or `pm` | lowercase meridiem indicator without periods
`A.M.` or `P.M.` | uppercase meridiem indicator with periods
`a.m.` or `p.m.` | lowercase meridiem indicator with periods 
````

**Review by benesch — COMMENTED** [review]
Thanks so much for writing this up, @quodlibetor!

File: doc/user/content/sql/functions/to_char.md (hunk 9 above)
Comment (quodlibetor) [reply doc/user/content/sql/functions/to_char.md:16]: Can do, it just looks weird using this as an h3.

**Comment by benesch** [comment]
Is it too big? Sean made all the headers bigger recently and I think there
have been some resulting casualties of formatting. Maybe h4 would work?

File: doc/user/content/sql/functions/to_char.md (hunk 9 above)
Comment (quodlibetor) [reply doc/user/content/sql/functions/to_char.md:16]: For reference, with just the bold:

image

I'll update this comment with screenshots of the other headings. I think I tried down to h4 and they all looked weird to me, but we can look.

Comment (benesch) [reply doc/user/content/sql/functions/to_char.md]: So it seems to me that the heading styling needs tweaking if we're abandoning semantics for the sake of formatting.  🤷 not worth holding up this PR over at all!

**Review by benesch — APPROVED** [review]

File: doc/user/content/sql/functions/to_char.md (hunk 9 above)
Comment (quodlibetor) [reply doc/user/content/sql/functions/to_char.md]: I agree, I opted for semantics over style, we can change it later.

**Pull request merged**